### PR TITLE
Miscellaneous selection cleanups

### PR DESCRIPTION
**Pull request by arvind**
Four commits to address four minor issues with selections.

### CODE REVIEW

File: src/compile/selection/interval.ts [hunk 1]
```diff
@@ -118,14 +118,22 @@ const interval:SelectionCompiler = {
       name: undefined,
       type: 'rect',
       encode: {
-        enter: {fill: {value: '#eee'}},
+        enter: {
```
Comment (kanitw) [inline src/compile/selection/interval.ts]: I guess we should make brush style configurable (in a later PR) ?  

We can just output some role name and use the role config. -- Given that we will output config in the Vega output soon in https://github.com/vega/vega-lite/pull/2372 -- This should be easily to implement.  

File an issue?

File: src/compile/selection/interval.ts [hunk 2]
```diff
@@ -118,14 +118,22 @@ const interval:SelectionCompiler = {
...
+          stroke: undefined
+        },
         update: update
       }
     }].concat(marks, {
       name: name + BRUSH,
       type: 'rect',
       encode: {
-        enter: {fill: {value: 'transparent'}},
+        enter: {
```
Comment (kanitw) [inline src/compile/selection/interval.ts]: This maybe a dumb question, but I wonder why we need two separate rects here for fill and stroke. (Maybe add a comment?)

File: src/compile/selection/interval.ts (hunk 1 above)
Comment (kanitw) [inline src/compile/selection/interval.ts]: For the implementation, after we merge #2372, I think we can make these brush have some role names and add role configs to the `DEFAULT_CONFIG` in `config.ts` for these roles  and it should work.

Comment (arvind) [inline src/compile/selection/interval.ts]: Done #2378.

File: src/compile/selection/interval.ts (hunk 2 above)
Comment (arvind) [inline src/compile/selection/interval.ts]: To ensure that the fill color of the brush doesn't interfere with the marks/grid lines/etc. but still allow users to interact with the brush. Will add a comment.

**Pull request merged**